integrationtest/backend/lucene/src/test/java/org/hibernate/search/integrationtest/backend/lucene/vector/LuceneVectorFieldIT.java

```diff
@@ -0,0 +1,464 @@
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import org.apache.lucene.document.Document;
+
+class LuceneVectorFieldIT {
+
+	private static final byte[] BYTE_VECTOR_1 = new byte[] { 1, 2, 3, 4 };
+	private static final byte[] BYTE_VECTOR_2 = new byte[] { 1, 1, 1, 1 };
+
+	private static final float[] FLOAT_VECTOR_1 = new float[] { 1.0f, 2.0f, 3.0f, 4.0f, 1.0f, 2.0f, 3.0f, 4.0f };
+	private static final float[] FLOAT_VECTOR_2 = new float[] { 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f };
+
+	@RegisterExtension
+	public final SearchSetupHelper setupHelper = SearchSetupHelper.create();
+
+	@Test
+	void simpleVectorSavedAndRetrieved() {
+		SimpleMappedIndex<IndexBinding> index = SimpleMappedIndex.of( IndexBinding::new ).name( "index" );
+		setupHelper.start().withIndexes( index ).setup();
+		initDataSimple( index );
+
+		SearchQuery<Document> query = index.createScope().query()
+				.select(
+						f -> f.extension( LuceneExtension.get() ).document()
+				)
+				.where( f -> f.matchAll() )
+				.toQuery();
+
+		List<Document> result = query.fetchAll().hits();
+		assertThat( result )
+				.hasSize( 2 )
+				.satisfies( containsDocument(
+						doc -> doc.hasField( "string", "keyword1" )
+								.hasVectorField( "byteVector", BYTE_VECTOR_1 )
+								.hasVectorField( "floatVector", FLOAT_VECTOR_1 )
+								.andOnlyInternalFields()
+				) )
+				.satisfies( containsDocument(
+						doc -> doc.hasField( "string", "keyword2" )
+								.hasVectorField( "byteVector", BYTE_VECTOR_2 )
+								.hasVectorField( "floatVector", FLOAT_VECTOR_2 )
+								.andOnlyInternalFields()
+				) );
+	}
+
+	@Test
+	void simpleVectorSavedAndRetrievedViaProjection() {
+		SimpleMappedIndex<IndexBinding> index = SimpleMappedIndex.of( IndexBinding::new ).name( "index" );
+		setupHelper.start().withIndexes( index ).setup();
+		initDataSimple( index );
+
+		SearchQuery<Object[]> query = index.createScope().query()
+				.select(
+						f -> f.composite().from(
+								f.field( "string" ),
+								f.field( "byteVector" ),
+								f.field( "floatVector" )
+						).asArray()
+				)
+				.where( f -> f.matchAll() )
+				.toQuery();
+
+		List<Object[]> result = query.fetchAll().hits();
+		assertThat( result )
+				.hasSize( 2 )
+				.containsOnly(
+						new Object[] { "keyword1", BYTE_VECTOR_1, FLOAT_VECTOR_1 },
+						new Object[] { "keyword2", BYTE_VECTOR_2, FLOAT_VECTOR_2 }
+				);
+	}
+
+	@Test
+	void simpleVectorPredicateNoFilter() {
+		SimpleMappedIndex<IndexBinding> index = SimpleMappedIndex.of( IndexBinding::new ).name( "index" );
+		setupHelper.start().withIndexes( index ).setup();
+		initDataSimpleNoFilter( index );
+
+		assertThatQuery(
+				index.createScope().query()
+						.select( f -> f.field( "byteVector" ) )
+						.where( f -> f.knn( 3 ).field( "byteVector" )
+								.matching( bytes( 4, (byte) 5 ) ) )
+		).hasTotalHitCount( 3 )
+				.hasHitsAnyOrder(
```

Comment:
Why "any" order?

Wouldn't `bytes(4, 5)` be expected to be the first hit, since it's identical to the requested vector?

Reply:
Good point 😃